aten/src/ATen/native/SobolEngineOps.cpp

```diff
@@ -150,7 +150,11 @@ Tensor& _sobol_engine_initialize_state_(Tensor& sobolstate, int64_t dimension) {
     }
   }
 
-  Tensor pow2s = at::pow(2, at::native::arange((MAXBIT - 1), -1, -1, sobolstate.options()));
+  Tensor pow2s = at::pow(2, at::native::arange(
+    (MAXBIT - 1), -1, -1,
+    typeMetaToScalarType(sobolstate.options().dtype()),
```

Comment:
This indentation is hella misleading. typeMetaToScalarType only has one argument, not three.